spec/services/punches_service_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe PunchesService do
+  let(:company) { create(:company) }
+  let(:user) { create(:user, :with_token, company: company) }
+  let(:project) { create(:project, company: user.company) }
+
+  describe "#create" do
+    context "with right arguments" do
+      let(:punches) {
+        [
+          {
+            "from" => "2022-04-19T12:00:00",
+            "to" => "2022-04-19T15:00:00",
+            "project_id" => "#{project.id}"
+          },
+          {
+            "from" => "2022-04-19T16:00:00",
+            "to" => "2022-04-19T21:00:00",
+            "project_id" => "#{project.id}"
+          }
+        ]
+      }
+
+      let(:office) { create(:office, :with_holiday) }
+
+      it "returns the created punches" do
```

Comment:
What do you think about changing this to something like?
```ruby
it "returns the created punches" do
  result = described_class.create(punches, user)

  # You may need to something like `result.map(&:to_json)`
  expect(result).to match_array(punches)
end
```

Reply:
I will refactor this